logic/transactionPool.js

```diff
@@ -276,9 +271,11 @@ TransactionPool.prototype.countUnconfirmed = function () {
  * @param {transaction} transaction
  */
 TransactionPool.prototype.addBundledTransaction = function (transaction) {
-	self.bundled.transactions.push(transaction);
-	var index = self.bundled.transactions.indexOf(transaction);
-	self.bundled.index[transaction.id] = index;
+	if (self.bundled.index[transaction.id] === undefined) {
+		self.bundled.transactions.push(transaction);
+		var index = self.bundled.transactions.indexOf(transaction);
```

Comment:
Nop, it doesn't works, should be replaced by `self.bundled.transactions.findIndex(function(tr) { return tr["id"] === transaction.id })`